### PR TITLE
Upgrade OSGi service to compatible with REST API

### PR DESCRIPTION
**Pull request by Thumimku**
## Purpose
Improve OSGI service compatible with the REST API service.

- Implement changes in OSGi service level, DAO, test classes, and DB scripts.

### CODE REVIEW

File: components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/RemoteFetchConfigurationService.java [hunk 1]
```diff
@@ -52,20 +52,22 @@ ValidationReport updateRemoteFetchConfiguration(RemoteFetchConfiguration fetchCo
...
      * This method is used to call by clients to get existing RemoteFetchConfiguration by Id.
      * This method is called by clients while triggering or edit existing RemoteFetchConfiguration.
-     * @param fetchConfigurationId ID
+     * @param fetchConfigurationId ID.
+     * @param tenantDomain Tenant Domain.
      * @return Remote Fetch Configuration for id.
      * @throws RemoteFetchCoreException
      */
-    RemoteFetchConfiguration getRemoteFetchConfiguration(String fetchConfigurationId)
+    RemoteFetchConfiguration getRemoteFetchConfiguration(String fetchConfigurationId, String tenantDomain)
```
[the remark this answers is not in the document]
Comment (ruwanta) [reply components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/RemoteFetchConfigurationService.java:60]: Having the tenant in. the DAO class is correct BTW

File: components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/RemoteFetchConfiguration.java [hunk 2]
```diff
@@ -28,8 +28,9 @@
  * Used by DAO to map configuration Db row data.
  * Used auto pull schedulers at runtime.
  */
-public class RemoteFetchConfiguration {
+public class RemoteFetchConfiguration implements java.io.Serializable {
```
[the remark this answers is not in the document]
Comment (Thumimku) [reply components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/RemoteFetchConfiguration.java:31]: While update remote fetch config via REST, 
- we first get the domain object from osgi get service.
- Then we need to dummy object from domain object then patch requested param.
- Then updated dummy will be send to osgi update service.

To create dummy object, i used object(input/output) stream and bytearray(input/output) stream. For the corresponding class should be serilizable.

[the remark this answers is not in the document]
Comment (ruwanta) [reply components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/RemoteFetchConfiguration.java:31]: For that, use different mechanisms
1. new RemoteFetchConfiguration()
2. BeanUtils.copyProperties(...) etc.

Using java serialization is waste of Memory and CPU resources.

[the remark this answers is not in the document]
Comment (Thumimku) [reply components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/RemoteFetchConfiguration.java]: ACK. resolved

File: components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/RemoteFetchConfigurationService.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Thumimku) [reply components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/RemoteFetchConfigurationService.java]: ACK. Resolved

**Pull request merged**